cmd/server/temporal/server.go

```diff
@@ -135,7 +135,11 @@ func (s *server) startService() common.Daemon {
 	servicePortMap := make(map[string]int)
 	for roleName, svcCfg := range s.cfg.Services {
 		serviceName := getServiceName(roleName)
-		servicePortMap[serviceName] = svcCfg.RPC.Port
+		if serviceName == common.FrontendServiceName || serviceName == common.MatchingServiceName {
```

Comment:
For some reason it worked for frontend even before this change.

Reply:
Probably no one role directly calls frontend